Add a PWM section in the i.MX porting guide (#7)
ms-iot/imx-iotcore: Documentation/porting-imx.md

```c
Package (2) { "Regulator1V8Exist", 0 }, // 1.8V Switching External Circuitry: Not implemented
```
### PWM

> For best experience using the PWM WinRT APIs from UWP apps, some device properties may need to be set where some of which are generally required for UWP access to the PWM device interface while others are required for some PWM WinRT APIs to function correctly. More on these device interface properties can be found here in the [Setting device interface section](https://docs.microsoft.com/en-us/windows-hardware/drivers/spb/pulse-width-controller%20driver#setting-device-interface-properties) of the PWM DDI MSDN documentation.

Reviewer: ```suggestion
For the best experience using the PWM WinRT APIs from UWP apps some additional device properties need to be set. Documentation on these device interface properties can be found here in the [Setting device interface properties](https://docs.microsoft.com/en-us/windows-hardware/drivers/spb/pulse-width-controller%20driver#setting-device-interface-properties) section of the PWM DDI MSDN article.
```

> For an example of setting the PWM deivce interface properties statically from an inf file, see the [Virtual PWM driver sample [Virtual PWM driver sample here](https://github.com/Microsoft/Windows-iotcore-samples/tree/develop/Drivers/VirtualPWM).

Reviewer 2: ```suggestion
For an example of setting the PWM deivce interface properties statically from an inf file, see the [Virtual PWM driver sample](https://github.com/Microsoft/Windows-iotcore-samples/tree/develop/Drivers/VirtualPWM).
```

> For example on setting the PWM deivce interface properties statically from an inf file, see the [Virtual PWM driver sample here](https://github.com/Microsoft/Windows-iotcore-samples/tree/develop/Drivers).

Reviewer: ```suggestion
For an example of setting the PWM deivce interface properties statically from an inf file, see the [Virtual PWM driver sample](https://github.com/Microsoft/Windows-iotcore-samples/tree/develop/Drivers).
```

Reviewer: We shouldn't merge this pull until the virtual PWM driver has been checked into Windows-iotcore-samples/Drivers, and this URL should be updated directly to its folder.

> For an example of how to read the ACPI _DSD from within a WDF driver and how you can assign the PWM device properties at runtime, see the [i.MX SDHC driver here](https://github.com/MHesham/imx-iotcore/tree/public_preview/driver/sd/imxusdhc).

Reviewer: ```suggestion
For an example of how to read the ACPI _DSD from within a WDF driver and how you can assign the PWM device properties at runtime, see the [i.MX SDHC driver here](../driver/sd/imxusdhc).
```

> For example on how to read ACPI _DSD in general from within a WDF driver and assigning so you can assign the PWM device properties in runtime, see the [i.MX SDHC driver here](https://github.com/MHesham/imx-iotcore/tree/public_preview/driver/sd/imxusdhc).

Reviewer: This should be a relative link to /driver/sd/imxusdhc, not an absolute link to your fork.

Reviewer: @christopherco Didn't we remove the _DSD dependency from the imxusdhc stack?

https://github.com/ms-iot/imx-edk2-platforms/blob/imx/Platform/SolidRun/HummingBoardEdge_iMX6Q_2GB/AcpiTables/Dsdt-Sdhc.asl

> For example on how to read ACPI _DSD in general from within a WDF driver and assigning so you can assign the PWM device properties in runtime, see the [i.MX SDHC driver here](https://github.com/MHesham/imx-iotcore/tree/public_preview/driver/sd/imxusdhc).

Reviewer: ```suggestion
For an example of how to read the ACPI _DSD from within a WDF driver and how you can assign the PWM device properties at runtime, see the [i.MX SDHC driver here](https://github.com/MHesham/imx-iotcore/tree/public_preview/driver/sd/imxusdhc).
```

> For example on how to read ACPI _DSD in general from within a WDF driver and assigning so you can assign the PWM device properties in runtime, see the [i.MX SDHC driver here](https://github.com/MHesham/imx-iotcore/tree/public_preview/driver/sd/imxusdhc).

Reviewer: ```suggestion
For an example of how to read the ACPI _DSD from within a WDF driver and how you can assign the PWM device properties at runtime, see the [i.MX SDHC driver here](https://github.com/MHesham/imx-iotcore/tree/public_preview/driver/sd/imxusdhc).
```
How does the SDHC driver relate to assigning PWM device properties at runtime?

Author: You are correct, the wording was wrong.

> For example on how to read ACPI _DSD in general from within a WDF driver and assigning so you can assign the PWM device properties in runtime, see the [i.MX SDHC driver here](https://github.com/MHesham/imx-iotcore/tree/public_preview/driver/sd/imxusdhc).

Reviewer: @christopherco Didn't we remove the _DSD dependency from the imxusdhc stack?

https://github.com/ms-iot/imx-edk2-platforms/blob/imx/Platform/SolidRun/HummingBoardEdge_iMX6Q_2GB/AcpiTables/Dsdt-Sdhc.asl

Author: We did, but the code that deals with ACPI _DSD and _DSM is still there in the usdhc code, but it is dormant at the moment. Should be enabled again once we fix the SD power issues. Other aspects of the _DSD/_DSM needs some rework to be standard compliant, but the acpiutil code is still a good reference.

Reviewer 2: The problem with any sort of _DSD/_DSM we have previously implemented is that the definitions/behaviors are not formally documented in a specification (Github readmes or code comments do not count). Upstream edk2 requires references a public documents for these types of OS-UEFI interactions.

#### PWM References:

- [PWM DDI](https://docs.microsoft.com/en-us/windows-hardware/drivers/spb/pulse-width-controller%20driver).
- [PWM Driver Reference](https://docs.microsoft.com/en-us/windows/desktop/devio/pwm-api).
- [PWM WinRT APIs](https://docs.microsoft.com/en-us/uwp/api/windows.devices.pwm).

> - [Virtual PWM Driver Sample](https://github.com/Microsoft/Windows-iotcore-samples/tree/develop/Drivers).

Reviewer: No need for periods after these links. They aren't sentences.
